File: cafe-management/com.inn.cafe/src/main/java/com/inn/cafe/service/BillService.java
package com.inn.cafe.service;

import com.inn.cafe.POJO.Bill;
import com.inn.cafe.POJO.DraftOrder;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

public interface BillService {
    ResponseEntity<String> generateReport(Map<String, Object> requestMap);

    ResponseEntity<List<Bill>> getBills();

    ResponseEntity<byte[]> getPdf(Map<String, Object> requestMap);

    ResponseEntity<String> deleteBill(Integer id);

    ResponseEntity<String> saveDraft(Map<String, Object> requestMap);

    ResponseEntity<List<DraftOrder>> getDrafts();

    ResponseEntity<DraftOrder> getDraftById(Integer id);

    ResponseEntity<String> deleteDraft(Integer id);
}

File: cafe-management/Frontend/src/app/services/bill.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class BillService {

  url = environment.apiUrl;

  constructor(private httpClient: HttpClient) { }
  
  generateReport(data:any){
    return this.httpClient.post(this.url+"/bill/generateReport", data, {
      headers: new HttpHeaders().set('Content-Type', "application/json")
    })
  }

  getPdf(data:any):Observable<Blob>{
    return this.httpClient.post(this.url+"/bill/getPdf", data, {responseType:'blob'});
  }

  getBills(){
    return this.httpClient.get(this.url+"/bill/getBills");
  }

  delete(id:any){
    return this.httpClient.post(this.url+"/bill/delete/"+id, {
      headers: new HttpHeaders().set('Content-Type', "application/json")
    })
  }

  saveDraft(data:any){
    return this.httpClient.post(this.url+"/bill/draft/save", data, {
      headers: new HttpHeaders().set('Content-Type', "application/json")
    })
  }

  getDrafts(){
    return this.httpClient.get(this.url+"/bill/draft/list");
  }

  getDraftById(id:any){
    return this.httpClient.get(this.url+"/bill/draft/"+id);
  }

  deleteDraft(id:any){
    return this.httpClient.post(this.url+"/bill/draft/delete/"+id, {
      headers: new HttpHeaders().set('Content-Type', "application/json")
    })
  }

}
